### PR TITLE
Adding Logs and fixes for the eod behavior of dqe currently

### PR DESCRIPTION
**Pull request by tszwailam**
Not Finished - but creating a PR to accept comments and feedbacks

### CODE REVIEW

File: code/processes/dqc.q [hunk 1]
```diff
@@ -31,11 +31,14 @@ init:{
   .dqe.loadtimer'[.dqe.configtable];
 
   .dqe.tosavedown:()!();                                                                                        /- store i numbers of rows to be saved down to DB
-  if[.z.p>.eodtime.nextroll:.eodtime.getroll[.z.p];system"t 0";'"next roll is in the past"]                     /- Checking if .eodtime.nextroll is correct
+  if[(`timestamp$.dqe.currentpartition)>=.eodtime.nextroll;.eodtime.getroll[`timestamp$.dqe.currentpartition];.lg.o[`dqc;"Manually update .eodtime.nextroll as it was incorrect"]]
```
Comment (ThomasSmyth) [inline code/processes/dqc.q:34]: I think this should be in `.u.end` as the current partition isn't updated until after `init` is called. This should be reverted back to the previous version (check against `.z.p`/`.proc.cp[]`). This will probably fall victim to the same issue as before.

File: code/processes/dqc.q [hunk 2]
```diff
@@ -31,11 +31,14 @@ init:{
...
+  /- Checking if .eodtime.nextroll is correct
+  .lg.o[`dqc;(".eodtime.nextroll set to ",string .eodtime.nextroll)];
```
Comment (ThomasSmyth) [inline code/processes/dqc.q:36]: Can remove brackets:
```suggestion
  .lg.o[`dqc;".eodtime.nextroll set to ",string .eodtime.nextroll];
```

File: code/processes/dqe.q [hunk 3]
```diff
@@ -17,9 +17,12 @@ init:{
   .timer.once[.eodtime.nextroll;(`.u.end;.dqe.getpartition[]);"Running EOD on Engine"];                         /- set timer to call EOD
   .dqe.tosavedown:()!();                                                                                        /- store i numbers of rows to be saved down to DB
   .dqe.configtimer[];
+  if[(`timestamp$.dqe.currentpartition)>=.eodtime.nextroll;.eodtime.getroll[`timestamp$.dqe.currentpartition];.lg.o[`dqc;"Manually update .eodtime.nextroll as it was incorrect"]]
+  .lg.o[`dqe;(".eodtime.nextroll set to ", string .eodtime.nextroll)];
```
Comment (ThomasSmyth) [inline code/processes/dqe.q:21]:
```suggestion
  .lg.o[`dqe;".eodtime.nextroll set to ", string .eodtime.nextroll];
```

File: code/processes/dqe.q [hunk 4]
```diff
@@ -81,6 +86,7 @@ writedownengine:{
   .timer.removefunc'[exec funcparam from .timer.timer where `.dqe.writedownengine in' funcparam];               /- clear writedown timer
   .dqe.init[];
   .dqe.currentpartition:pt+1;
+  .lg.o[`dqe;".u.end finished"];
   };
 
-.dqe.init[]
+.dqe.init[[]]
```
Comment (ThomasSmyth) [inline code/processes/dqe.q:92]: Revert back.

File: code/processes/dqe.q [hunk 5]
```diff
@@ -5,7 +5,7 @@ gmttime:@[value;`gmttime;1b];
 partitiontype:@[value;`partitiontype;`date];
 getpartition:@[value;`getpartition;
   {{@[value;`.dqe.currentpartition;
-    (`date^partitiontype)$(.z.D,.z.d)gmttime]}}];
+    (`date^partitiontype)$([].z.D,.z.d)gmttime]}}];
```
Comment (ThomasSmyth) [inline code/processes/dqe.q:8]: Is this change required?

Comment (tszwailam) [reply code/processes/dqe.q:8]: no, will revert back a lot of those

File: code/processes/dqc.q [hunk 6]
```diff
@@ -31,11 +31,13 @@ init:{
   .dqe.loadtimer'[.dqe.configtable];
 
   .dqe.tosavedown:()!();                                                                                        /- store i numbers of rows to be saved down to DB
-  if[.z.p>.eodtime.nextroll:.eodtime.getroll[.z.p];system"t 0";'"next roll is in the past"]                     /- Checking if .eodtime.nextroll is correct
+  /- Checking if .eodtime.nextroll is correct
+  .lg.o[`dqc;".eodtime.nextroll set to ",string .eodtime.nextroll];
```
Comment (ThomasSmyth) [inline code/processes/dqc.q:35]: Move this to where `nextroll` is updated in `.u.end`

File: code/processes/dqc.q [hunk 7]
```diff
@@ -219,10 +224,13 @@ reruncheck:{[chkid]
   .timer.removefunc'[exec funcparam from .timer.timer where `.dqe.writedown in' funcparam];                     /- clear writedown timer
   .timer.removefunc'[exec funcparam from .timer.timer where `.dqe.writedownconfig in' funcparam];               /- clear writedownconfig timer
   delete configtable from `.dqe;
-  .dqe.init[];
   .dqe.currentpartition:pt+1;
+  if[(`timestamp$.dqe.currentpartition)>=.eodtime.nextroll;.eodtime.getroll[`timestamp$.dqe.currentpartition];.lg.o[`dqc;"Manually update .eodtime.nextroll as it was incorrect"]]
+  .dqe.init[];
```
Comment (ThomasSmyth) [inline code/processes/dqc.q:229]: Can move this back as it has no more reliance on the partition.

File: code/processes/dqc.q [hunk 8]
```diff
@@ -219,10 +224,13 @@ reruncheck:{[chkid]
   .timer.removefunc'[exec funcparam from .timer.timer where `.dqe.writedown in' funcparam];                     /- clear writedown timer
   .timer.removefunc'[exec funcparam from .timer.timer where `.dqe.writedownconfig in' funcparam];               /- clear writedownconfig timer
   delete configtable from `.dqe;
-  .dqe.init[];
   .dqe.currentpartition:pt+1;
+  if[(`timestamp$.dqe.currentpartition)>=.eodtime.nextroll;.eodtime.getroll[`timestamp$.dqe.currentpartition];.lg.o[`dqc;"Manually update .eodtime.nextroll as it was incorrect"]]
```
Comment (ThomasSmyth) [inline code/processes/dqc.q:228]: Comment is a bit clunky. Needs to say about moving `nextroll` to the next day. Something like "Moving nextroll to match current partition".

File: code/processes/dqe.q [hunk 9]
```diff
@@ -5,7 +5,7 @@ gmttime:@[value;`gmttime;1b];
 partitiontype:@[value;`partitiontype;`date];
 getpartition:@[value;`getpartition;
   {{@[value;`.dqe.currentpartition;
-    (`date^partitiontype)$(.z.D,.z.d)gmttime]}}];
+    (`date^partitiontype(.z.D,.z.d)gmttime]}}];
```
Comment (ThomasSmyth) [inline code/processes/dqe.q:8]: I think you've broke this line :eyes: missing a bracket somewhere.

File: code/processes/dqc.q [hunk 10]
```diff
@@ -219,10 +222,14 @@ reruncheck:{[chkid]
   .timer.removefunc'[exec funcparam from .timer.timer where `.dqe.writedown in' funcparam];                     /- clear writedown timer
   .timer.removefunc'[exec funcparam from .timer.timer where `.dqe.writedownconfig in' funcparam];               /- clear writedownconfig timer
   delete configtable from `.dqe;
-  .dqe.init[];
   .dqe.currentpartition:pt+1;
+  if[(`timestamp$.dqe.currentpartition)>=.eodtime.nextroll;.eodtime.nextroll:.eodtime.getroll[`timestamp$.dqe.currentpartition];.lg.o[`dqc;"Moving .eodtime.nextroll to match current partition"]]
```
Comment (ThomasSmyth) [inline code/processes/dqc.q:226]: For the sake of readability split this up + add a semi colon to the end as it is missing + add a comment to explain why this statement exists:
```suggestion
  if[(`timestamp$.dqe.currentpartition)>=.eodtime.nextroll;
    .eodtime.nextroll:.eodtime.getroll[`timestamp$.dqe.currentpartition];
    .lg.o[`dqc;"Moving .eodtime.nextroll to match current partition"]
  ];
```

File: code/processes/dqe.q [hunk 11]
```diff
@@ -73,14 +75,18 @@ writedownengine:{
...
+  .lg.o[`dqe;".u.end initiated"];
   .dqe.endofday[.dqe.dqedbdir;.dqe.getpartition[];`resultstab;`.dqe;.dqe.tosavedown[`.dqe.resultstab]];
   hdbs:distinct raze exec w from .servers.SERVERS where proctype=`dqedb;                                        /- get handles for DBs that need to reload
   .dqe.notifyhdb[.os.pth .dqe.dqedbdir]'[hdbs];                                                                 /- send message for DBs to reloadi
   .timer.removefunc'[exec funcparam from .timer.timer where `.dqe.runquery in' funcparam];                      /- clear check function timers
   .timer.removefunc'[exec funcparam from .timer.timer where `.u.end in' funcparam];                             /- clear EOD timer
   .timer.removefunc'[exec funcparam from .timer.timer where `.dqe.writedownengine in' funcparam];               /- clear writedown timer
-  .dqe.init[];
   .dqe.currentpartition:pt+1;
+  if[(`timestamp$.dqe.currentpartition)>=.eodtime.nextroll;.eodtime.nextroll:.eodtime.getroll[`timestamp$.dqe.currentpartition];.lg.o[`dqe;"Moving .eodtime.nextroll to match current partition"]]
```
Comment (ThomasSmyth) [inline code/processes/dqe.q:86]: Same here comments + semi colon + split over multiple lines:
```suggestion
  if[(`timestamp$.dqe.currentpartition)>=.eodtime.nextroll;
    .eodtime.nextroll:.eodtime.getroll[`timestamp$.dqe.currentpartition];
    .lg.o[`dqe;"Moving .eodtime.nextroll to match current partition"]
  ];
```
